aurmr_tasks/scripts/aurmr_demo

```diff
@@ -65,159 +59,178 @@ class RecordTime(State):
+            #                     {"succeeded": "PICTURE_WAIT", "aborted": "MOVE_INTO_JOINT_LIMITS"})
+
             StateMachine.add_auto("MOVE_TO_PRE_TOTE_POSE", motion.MoveToJointAngles(robot, "drop_hide"), ["succeeded", "aborted"])
-            StateMachine.add_auto("PICTURE_WAIT", Wait(2), ["succeeded"])
             StateMachine.add("CHECK_FOR_GRIPPER_ITEM", motion.CheckGripperItem(robot),
-                            {"item_detected": "POST_PICK_UPDATE_MASKS", "no_item_detected": "UPDATE_BIN_MASKS"})
-            StateMachine.add("POST_PICK_UPDATE_MASKS", perception.PickObject(),
-                            {"succeeded": "PUBLISH_PICK_STATUS", "aborted": "PUBLISH_PICK_STATUS", "preempted": "PUBLISH_PICK_STATUS"})
-            StateMachine.add("UPDATE_BIN_MASKS", perception.UpdateBin(),
-                            {"succeeded": "USER_PROMPT_FOR_RETRY", "aborted": "USER_PROMPT_FOR_RETRY", "preempted": "USER_PROMPT_FOR_RETRY"})
-            StateMachine.add('USER_PROMPT_FOR_RETRY', UserPromptForRetry(robot.tf2_buffer, timeout_connection_secs=1.0), {'retry': 'CLEAR_SCENE_PREMOVE', 'continue': 'PUBLISH_PICK_STATUS'})
-            StateMachine.add_auto("PUBLISH_PICK_STATUS", PublishStatus(status_pub), ["succeeded"])
-            StateMachine.add("ASK_FOR_GRIPPER_OPEN_TOTE_RELEASE", motion.OpenGripper(robot, return_before_done=True), {"succeeded": "BLOW_OFF_GRIPPER", "aborted": "BLOW_OFF_GRIPPER", "preempted": "BLOW_OFF_GRIPPER"})
-            StateMachine.add("BLOW_OFF_GRIPPER", motion.BlowOffGripper(robot, return_before_done=True), {"succeeded": "done", "aborted": "done", "preempted": "done"})
+                            {"item_detected": "POST_PICK_SUCCESS_CLEANUP", "no_item_detected": "RETRY"})
+
+            
+            with pick_success_cleanup_sm:
+                StateMachine.add_auto("BLOW_OFF_GRIPPER", motion.BlowOffGripper(robot, return_before_done=True), ["succeeded", "aborted", "preempted"])
+                StateMachine.add_auto("PICTURE_WAIT", Wait(2), ["succeeded"])
+                StateMachine.add_auto("POST_PICK_UPDATE_MASKS", perception.PickObject(), ["succeeded", "preempted", "aborted"])
+                StateMachine.add("ASK_FOR_GRIPPER_OPEN_TOTE_RELEASE", motion.OpenGripper(robot, return_before_done=True), {"succeeded": "succeeded", "preempted": "succeeded", "aborted": "succeeded"})
+
+            with pick_retry_sm:
+                StateMachine.add_auto("BLOW_OFF_GRIPPER", motion.BlowOffGripper(robot, return_before_done=True), ["succeeded", "aborted", "preempted"])
+                StateMachine.add_auto("PICTURE_WAIT", Wait(2), ["succeeded"])   
+                StateMachine.add_auto("UPDATE_BIN_MASKS", perception.UpdateBin(), ["succeeded", "preempted", "aborted"])
+                StateMachine.add('USER_PROMPT_FOR_RETRY', UserPromptForRetry(robot.tf2_buffer, timeout_connection_secs=1.0), {"retry": "REMAP_GRASP", "continue": "continue"})
```

Comment:
Has this change been tested? It seems like this could cause problems for the retry.

After retry the state machine should goto `GET_GRASP_POSE`. It's important to goto that state because that is where we override `grasp_pose` from `human_grasp_pose` _and_ (importantly) we unset `human_grasp_pose`: https://github.com/au-rmr/aurmr_tahoma/blob/main/aurmr_tasks/src/aurmr_tasks/common/perception.py#L176-L179  It does not seem like this `cf.remap_auto` emulates that behavior.

Also `human_pre_grasp_pose` is new to me and doesn't seem to be set anywhere in the state machine -- but seems to be assumed to exist here?

Reply:
Have not tested yet the retry yet, but plan to tomorrow. I began changing the retry state's interface with the objective of matching the interface of the perception-based GetGraspPose state. I think we can have `UserPromptForRetry` write out directly to the appropriate keys, so the rest of the states can be ignorant to the substitution of the human help. If you take a look at `GetGraspPose` you can see that I already narrowed its responsibility to be just perception-based grasp pose detection. 

Depending on whether this path seems like its working, I was planning to try to insert another copy of GetGraspPose for when object segmentation completely fails. It was probably partially because of the outdated model weights, but we had to pass on several objects Friday because the state machine gives up when there are no masks.